### PR TITLE
Document ES Modules usage

### LINKED ISSUE #130: Support ES module scripts?
I've tried writing a silly deploy script as an ES module, now that node has experimental support, but meow doesn't grok this:

```
> yarn deploy
yarn run v1.21.0
$ node src/__scripts__/deploy.mjs
(node:5588) ExperimentalWarning: The ESM module loader is experimental.
myproj/node_modules/meow/index.js:15
const parentDir = path.dirname(module.parent.filename);
                                             ^

TypeError: Cannot read property 'filename' of undefined
    at Object.<anonymous> (myproj/node_modules/meow/index.js:15:46)
```

I'm curious of meow really needs to look up the parent module?
If it's part of meow's magic, is there perhaps a work-around?

In my case, my script is a tiny part of a larger project, and at present I can't / don't want to invoke meow on the top-level thing. Instead I'm trying to write a tiny script that only happens to be the directory structure of `myproj`.

Is `meow` not a good fit here, or am I just missing something?

### PR DESCRIPTION
**Pull request by jimthedev**
Closes #130.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you also add Node.js 14 here: https://github.com/sindresorhus/meow/blob/c67d9f4f8813d03a97637d8460274e63956cd096/.travis.yml#L3

**Comment by jimthedev** [comment]
@sindresorhus added 14 to Travis, also made the esm test conditional on node 14 and greater since otherwise it’ll throw an error. Let me know if there’s a different way you’d prefer to handle that one.

**Comment by jimthedev** [comment]
Ok RTG, @sindresorhus! Thanks for this project and all your projects.

**Comment by jimthedev** [comment]
Just tried this an ran into an issue. Don't merge for now.

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+estest/index.js
```
Comment (jimthedev) [inline .eslintignore:1]: We must ignore this file due to https://github.com/eslint/eslint/issues/13133

**Comment by jimthedev** [comment]
Ok I updated this one (which is now ready to review @sindresorhus). According to https://twitter.com/MylesBorins/status/1255950295806488576 it looks like basically the user must use createRequire if they want to keep using this in ES modules (but they are doing so using commonjs). Seems ok for now?

I am open to other ideas on how to accomplish this but at least it would unblock people looking to use it in their ES based cli's.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**